src/neighbors.cc

```diff
@@ -1421,25 +1421,16 @@ dump_peer_options(StoreEntry * sentry, CachePeer * p)
 #if USE_HTCP
     if (p->options.htcp) {
         os << " htcp";
-        if (p->options.htcp_oldsquid || p->options.htcp_no_clr || p->options.htcp_no_purge_clr || p->options.htcp_only_clr) {
-            bool doneopts = false;
-            if (p->options.htcp_oldsquid) {
-                os << (doneopts ? ',' : '=') << "oldsquid";
-                doneopts = true;
-            }
-            if (p->options.htcp_no_clr) {
-                os << (doneopts ? ',' : '=') << "no-clr";
-                doneopts = true;
-            }
-            if (p->options.htcp_no_purge_clr) {
-                os << (doneopts ? ',' : '=') << "no-purge-clr";
-                doneopts = true;
-            }
-            if (p->options.htcp_only_clr) {
-                os << (doneopts ? ',' : '=') << "only-clr";
-                //doneopts = true; // uncomment if more opts are added
-            }
-        }
+        std::vector<const char *> opts{};
```

Comment:
Please add an allocator for this container. I think we can live with this new memory allocation/deallocation, but we should direct _all_ dynamic allocations through our memory allocating code. In this particular case, doing so will not save many cycles, but it is a good idea for other (obvious?) reasons.

Something like this might work (but may require an `#include`):

```suggestion
        std::vector<const char *, PoolingAllocator<const char *> > opts;
```